File: app/src/main/java/com/autoscout24/carfinder/ui/utils/MathUtils.java
package com.autoscout24.carfinder.ui.utils;

public class MathUtils {

    public static int mod(int x, int y) {
        int result = x % y;
        return result < 0? result + y : result;
    }

    public static int arraySizeRequiredForAdvertisements(int arraySize, int positionOfElement) {
        if(positionOfElement > 0) {
            return (int)arraySize+mod(arraySize,positionOfElement);
        }else return 0;


    }
}

File: app/src/main/java/com/autoscout24/carfinder/ui/utils/MathUtils.kt
package com.autoscout24.carfinder.ui.utils

object MathUtils {

    fun mod(x: Int, y: Int): Int {
        val result = x % y
        return if (result < 0) result + y else result
    }

    fun arraySizeRequiredForAdvertisements(arraySize: Int, positionOfElement: Int): Int {
        return if (positionOfElement > 0) {
            arraySize + mod(arraySize, positionOfElement)
        } else
            0


    }
}
